pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -0,0 +1,346 @@
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.prepareReactor1 != nil {
+				tt.prepareReactor1(tt.antreaClientset)
+			}
+			if tt.prepareReactor2 != nil {
+				tt.prepareReactor2(tt.k8sClientset)
+			}
+			clients, err := createAgentClients(tt.k8sClientset, tt.antreaClientset, clientConfig, "", nameList)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				agentClients := []string{}
+				for client := range clients {
+					agentClients = append(agentClients, client)
+				}
+				sort.Strings(agentClients)
+				assert.Equal(t, tt.expectedClients, agentClients)
+			}
+		})
+	}
+}
+
+func TestRequest(t *testing.T) {
+	k8sClientset := fake.NewSimpleClientset(&node1, &node3)
+	antreaClientset := fakeclientset.NewSimpleClientset(agentInfo1, agentInfo2, &controllerInfo)
+	agentClients, err := createAgentClients(k8sClientset, antreaClientset, clientConfig, "", nameList)
+	require.NoError(t, err)
+	controllerClient, err := createControllerClient(k8sClientset, antreaClientset, clientConfig)
+	require.NoError(t, err)
+	amount := len(agentClients)*2 + 2
+	bar := barTmpl.Start(amount)
+	defer bar.Finish()
+	defer bar.Set("prefix", "Finish ")
+	expectedResults := map[string]string{
+		"":       "Post \"http://172.16.0.0:0/supportbundle/system.antrea.io/v1beta1/supportbundles\": dial tcp 172.16.0.0:0:",
+		"node-1": "Post \"http://172.16.0.0:0/supportbundle/system.antrea.io/v1beta1/supportbundles\": dial tcp 172.16.0.0:0:",
+		"node-3": "Post \"http://127.0.0.1:0/supportbundle/system.antrea.io/v1beta1/supportbundles\": dial tcp 127.0.0.1:0:",
+	}
+	results := requestAll(agentClients, controllerClient, bar)
+	for node := range expectedResults {
+		assert.ErrorContains(t, results[node], expectedResults[node])
+	}
+}
+
+func TestDownload(t *testing.T) {
+	if option.dir == "" {
+		option.dir, _ = os.Getwd()
+	}
+	dir, err := filepath.Abs(option.dir)
+	require.NoError(t, err)
+	k8sClientset := fake.NewSimpleClientset(&node1, &node3)
+	antreaClientset := fakeclientset.NewSimpleClientset(agentInfo1, agentInfo2, &controllerInfo)
+	agentClients, err := createAgentClients(k8sClientset, antreaClientset, clientConfig, "", nameList)
+	require.NoError(t, err)
+	controllerClient, err := createControllerClient(k8sClientset, antreaClientset, clientConfig)
+	require.NoError(t, err)
+	amount := len(agentClients)*2 + 2
+	bar := barTmpl.Start(amount)
+	defer bar.Finish()
+	defer bar.Set("prefix", "Finish ")
+	resultMap := map[string]error{
+		"":       fmt.Errorf("error-0"),
+		"node-1": fmt.Errorf("error-1"),
+		"node-3": nil,
+	}
+	results := downloadAll(agentClients, controllerClient, dir, bar, resultMap)
+	expectedResults := map[string]string{
+		"":       `error-0`,
+		"node-1": `error-1`,
+		"node-3": "error when requesting the agent support bundle: Get \"http://127.0.0.1:0/supportbundle/system.antrea.io/v1beta1/supportbundles/agent\": dial tcp 127.0.0.1:0:",
```

Comment:
same comment as above: you should use https://pkg.go.dev/k8s.io/client-go@v0.26.3/rest/fake for all your REST clients.